Source/CBL_Replicator.m

```diff
@@ -341,13 +341,6 @@ - (void) start {
                  }
                 ];
 
-    // If client didn't set an authorizer, use basic auth if credential is available:
-    if (!_authorizer) {
-        _authorizer = [[CBLBasicAuthorizer alloc] initWithURL: _remote];
```

Comment:
We don't know if the authentication will be basic or digest from the url. Setting the basic authorizer from the URL here will cause CBLRemoteRequest and CBLSocketChangeTracker to always send the 1st request with Basic Authorization header. If the authentication happens to be digest, the password will be leaked in the header as the Basic authentication.

1.https://github.com/couchbase/couchbase-lite-ios/blob/feature/issue_777_digest_auth/Source/CBLRemoteRequest.m#L102-L108

2. https://github.com/couchbase/couchbase-lite-ios/blob/feature/issue_777_digest_auth/Source/ChangeTracker/CBLSocketChangeTracker.m#L80-L86